reana-ui/src/pages/workflowDetails/WorkflowDetails.js

```diff
@@ -32,20 +38,47 @@ export default function WorkflowDetailsPage() {
   );
 }
 
+const FINISHED_STATUSES = ["finished", "failed", "stopped", "deleted"];
+
 function WorkflowDetails() {
   const { id: workflowId } = useParams();
 
   const dispatch = useDispatch();
   const workflow = useSelector(getWorkflow(workflowId));
   const loading = useSelector(loadingWorkflows);
   const workflowsFetched = useSelector(isWorkflowsFetched);
+  const reanaToken = useSelector(getReanaToken);
+  const config = useSelector(getConfig);
+  const interval = useRef(null);
+  const { pollingSecs } = config;
 
   // TODO: add `workflowRefresh` dependency to trigger useEffect
   // once user opens, closes Interactive session, deletes workflow
   useEffect(() => {
     dispatch(fetchWorkflow(workflowId));
+    setPolling();
+    return cleanPolling;
   }, [dispatch, workflowId]);
 
+  useEffect(() => {
+    if (workflow && FINISHED_STATUSES.includes(workflow.status)) {
+      cleanPolling();
+    }
+  }, [workflow]);
+
+  const setPolling = () => {
+    if (!interval.current && reanaToken && pollingSecs) {
```

Comment:
Do we need the reana token here? I understand this was added to the workflow list page to prevent the polling when the token was still not granted but I don't think anyone can get to the workflow details page with no token.